### PR TITLE
K8s backend

### PR DESCRIPTION
**Pull request by Kelvin-M**
Non-mellody branch
Include:
- k8s-tasks
- security improvements
- fix concurrency

Companion PR :
https://github.com/SubstraFoundation/hlf-k8s/pull/86
https://github.com/SubstraFoundation/substra-tests/pull/126

Related : 
https://github.com/SubstraFoundation/substra-backend/pull/267
https://github.com/SubstraFoundation/substra-backend/pull/279

### CODE REVIEW

**Comment by inalgnu** [comment]
Before merging this PR:

- We have to elect one builder, having 3 builders was interesting to demonstrate the pros and cons of each of them but we won't provide support for multiple builders. I'm in favor of electing Kaniko for now.

- The code related to handling Makisu and DIND should be removed.

**Comment by Kelvin-M** [comment]
I don't understand why we have to choose for a builder. K8S docker image builders (kaniko, makisu, img, etc.) are in dev. and we don't know which one will be good in the future.

DinD is used for the e2e because it's faster (10m) than kaniko and makisu (more than 20m) which is good for a dev setup.

IMO, as the code for image builder is very light and simple they is no real support to do.

File: backend/substrapp/tasks/exceptions.json [hunk 1]
```diff
@@ -442,5 +442,8 @@
     "error": "0441",
     "gaierror": "0442",
     "herror": "0443",
-    "timeout": "0444"
+    "timeout": "0444",
+    "BadGzipFile": "0445",
+    "InvalidCacheKey": "0446",
+    "_GiveupOnFastCopy": "0447"
 }
```
Comment (inalgnu) [inline backend/substrapp/tasks/exceptions.json:449]: missing EOF

Comment (Kelvin-M) [reply backend/substrapp/tasks/exceptions.json]: Fixed by https://github.com/SubstraFoundation/substra-backend/pull/299/commits/eddd5980ca75fe31eb73b44a21bfbdd912af0cab

**Pull request merged**